omnikinverter/tcp.py

```diff
@@ -70,12 +70,100 @@ class _TcpData(BigEndianStructure):
         ("padding1", c_ubyte * 4),
         ("unknown0", c_ushort),
         ("padding2", c_ubyte * 10),
+    ]
+
+    @classmethod
+    def parse(cls: _TcpData, data: bytes, offset: int = 0) -> dict[str, Any]:
```

Comment:
`mypy` and `ruff` are really fighting over `classmethod`s: unsure how to solve their warnings or if this pattern is even desired.

Reply:
Ah we simply had to allow `ANN102` in addition to `ANN101`.